parser/SqlParser.ypp

```diff
@@ -1627,6 +1634,19 @@ literal_value:
       YYERROR;
     }
   }
+  | TOKEN_INTERVAL TOKEN_STRING_SINGLE_QUOTED datetime_unit {
+    quickstep::StringParseLiteralValue *parse_value;
+    const std::string& datetime_type_value = $3->value();
+    if (quickstep::StringParseLiteralValue::ParseAmbiguousInterval(&($2->append((" " +
+        datetime_type_value).c_str(), datetime_type_value.length() + 1)),
```

Comment:
Perhaps it's better to put this expression as the first argument to `ParseAmbiguousInterval` in one single line. I.e.
```
    if (quickstep::StringParseLiteralValue::ParseAmbiguousInterval(
        &($2->append((" " + datetime_type_value).c_str(), datetime_type_value.length() + 1)),
        &parse_value)) {
      $$ = parse_value;
    } else {
      ...
```

Also update `const std::string& datetime_type_value` to `const std::string &datetime_type_value`.